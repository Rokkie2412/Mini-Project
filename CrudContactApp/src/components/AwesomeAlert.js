/**
 * @format
 * @flow
 */
import * as React from 'react'
import Modal from "react-native-modal";
import {Pressable,Text,View} from 'react-native'
import type {PickAdd} from './component.type'
import {openGallery,openCamera} from '../functions/FunctionScreen'
import styles from '../styles/ModalStylePickImage'
import Icon from 'react-native-vector-icons/Ionicons'

const Awesome = ({show,setshow,image,setimage}:PickAdd):React.Node =>{
    return(
        <Modal
        isVisible={show}
        style={styles.mainContainer}>
            <Pressable
            testID='Close'
            onPress={()=>setshow(false)}
            >
                    <Icon name='close' style={styles.close}/>
            </Pressable>
            <Text style={styles.Header}>Choose image source :</Text>
            <View style={styles.iconView}>
                <Pressable
                testID='openCamera'
                onPress={()=>{
                    openCamera(image,setimage)
                    setshow(false)
                }}
                >
                    <Icon name='camera' style={styles.icon}/>
                </Pressable>
                <Pressable
                testID='openGallery'
                onPress={()=>{
                    openGallery(image,setimage)
                    setshow(false)
                }}
                >
                    <Icon name='image' style={styles.icon}/>
                </Pressable>
            </View>

        </Modal>
    )
}

export default Awesome