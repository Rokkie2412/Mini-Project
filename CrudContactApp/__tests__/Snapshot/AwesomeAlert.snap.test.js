import {create,act} from 'react-test-renderer';
import React from 'react';
import AwesomeAlert from "../../src/components/AwesomeAlert";


jest.mock('react-native-vector-icons/Ionicons', () => 'Mocked Ionicons');
jest.mock('react-native-modal',()=> "Mocked Modal React Native")
jest.mock('@react-native-community/netinfo',()=>'Mocked Netinfo')
jest.mock('react-native-image-picker',()=>{
    return{
        launchCamera:jest.fn(),
        launchImageLibrary:jest.fn()
    }
})
const func = jest.fn()
const tree = create(<AwesomeAlert setshow={func} setimage={func}/>)

describe('Testing for AwesomeAlert components', () => {
  describe('snapshot testing for AwesomeAlert', () => {
    it('Should get render correctly',()=>{
        expect(tree).toMatchSnapshot()
    })
  })
  describe('Testing button for awesomeAlert', () => {
    it('Should be pressed for cancel button',()=>{
        const cancel = tree.root.findByProps({testID:'Close'}).props
        act(()=>cancel.onPress())
    })
    it('Should be pressed for openCamera button',()=>{
        const camera = tree.root.findByProps({testID:'openCamera'}).props
        act(()=>camera.onPress())
    })
    it('Should be pressed for openCamera button',()=>{
        const Gall = tree.root.findByProps({testID:'openGallery'}).props
        act(()=>Gall.onPress())
    })
  })
  
  
})
