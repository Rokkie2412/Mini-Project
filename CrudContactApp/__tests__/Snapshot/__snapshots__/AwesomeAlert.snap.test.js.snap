// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for AwesomeAlert components snapshot testing for AwesomeAlert Should get render correctly 1`] = `
<Mocked Modal React Native
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#EEEEEE",
      "borderRadius": 30,
      "flex": 1,
      "height": "10%",
      "marginVertical": "50%",
      "position": "absolute",
      "width": "60%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="Close"
  >
    <Mocked Ionicons
      name="close"
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "#222831",
          "borderRadius": 50,
          "color": "#EEEEEE",
          "marginBottom": 20,
          "marginRight": 10,
          "padding": 8,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#222831",
        "fontSize": 18,
        "fontWeight": "600",
        "textAlign": "center",
      }
    }
  >
    Choose image source :
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="openCamera"
    >
      <Mocked Ionicons
        name="camera"
        style={
          Object {
            "backgroundColor": "#222831",
            "borderRadius": 50,
            "color": "#EEEEEE",
            "fontSize": 24,
            "margin": 10,
            "padding": "10%",
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="openGallery"
    >
      <Mocked Ionicons
        name="image"
        style={
          Object {
            "backgroundColor": "#222831",
            "borderRadius": 50,
            "color": "#EEEEEE",
            "fontSize": 24,
            "margin": 10,
            "padding": "10%",
          }
        }
      />
    </View>
  </View>
</Mocked Modal React Native>
`;
